### DISCUSSION TITLE
How can I manage my disk size in WSL2

### DISCUSSION THREAD

**Discussion by RayoBoi** [microsoft/WSL#6855] [Q&A]
**Is your feature request related to a problem? Please describe.**
I am new to wsl2 and programming in general, and I am in the basics. I'm just getting everything ready to start using wsl2 and the first thing I want to do is lower the amount of ram and disk it uses

**Describe the solution you'd like**
I would like to be able to have the ram and the disk well configured to start programming

**Describe alternatives you've considered**
I already managed to lower the ram from 12gb to 4gb, which is what I need for now, but I don't know how to lower the amount of disk that it will occupy

**Additional context**
If you have any idea or know how to do to put a specific amount of disk please tell me.

**Comment by 0xbadfca11** [comment]
See this first.
https://docs.microsoft.com/en-us/windows/wsl/compare-versions#expanding-the-size-of-your-wsl-2-virtual-hard-disk
Shrink filesystem with resize2fs first.
Then shrink the VHDX file with diskpart's shrink command.

**Comment by okibcn** [comment]
Sadly, WSL doesn't have a way to control the size of the vhdx virtual drive images. They grow uncontrolled way beyond the actual data contained.

For instance, you can see the actual data by typing `df /` inside your WSL distro. Maybe the [wslcompact](https://github.com/okibcn/wslcompact) script can help you in periodically compacting the size of your installed distros without having to purchase a 2 TB drive.
